LoopKitUI/Views/Deletable.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  Deletable.swift
+//  LoopKitUI
+//
+//  Created by Michael Pangburn on 4/30/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+
+enum TableDeletionState: Equatable {
+    case disabled
+    case enabled
+    case awaitingConfirmation(toDeleteItemAt: Int)
+
+    var isAwaitingConfirmation: Bool {
+        if case .awaitingConfirmation = self {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    var indexAwaitingDeletionConfirmation: Int? {
+        if case .awaitingConfirmation(toDeleteItemAt: let index) = self {
+            return index
+        } else {
+            return nil
+        }
+    }
+}
+
+
+/// Mimics the behavior of UITableViewCell deletion.
+struct Deletable<Content: View>: View {
```

Comment:
What is driving this reimplementation of List's built in delete mechanism? I think you mentioned this at our dev meeting, but maybe a short description of the rationale would be good recorded history here.

Reply:
Updated the comment here.
It boils down to:
1. SwiftUI's List is broken with resizing cells.
2. We created CardList to circumvent (1).
3. Since we aren't using List, we don't get built-in deletion for free.

Similarly to the navigation bar header, I'm not happy with replicating system functionality here. Jumping through hoops to get around broken SwiftUI API is very unsatisfying. 😞